### DISCUSSION TITLE
How can I update a user?

### DISCUSSION THREAD

**Discussion by aristotekean** [pocketbase/pocketbase#1671] [Q&A] [answered]
Hi I am trying update a user from my user collection, I used the js sdk and the angular http client, but ever I get a 404 response:

Request:
:authority:mydomainfly.dev
:method: PATCH
:path: /api/collections/users/records/3lz1zu8u8xqrdqx
content-type: application/json
and others ...

Payload:
{
    "name": "kevin bueno",
    "username": "aristotekean",
    "email": "jd@gmail.com"
}

Response:
{
    "code": 404,
    "message": "The requested resource wasn't found.",
    "data": {}
}

The flow is the following:
1. I as a normal user (not admin) to do login
2. I as a logged in user try update/complete my profile

any ideas why i received this code if I am do the stepts of the docs?
![image](https://user-images.githubusercontent.com/12689339/213957026-d8aab557-e27a-429c-bdae-d0c34ddb8a28.png)

**Comment by ganigeorgiev** [comment]
Without a code sample how you are sending the request, I'm not sure how to reproduce/debug it.
Instead of manually constructing it with the angular http client, you can also try sending it with the official [JS SDK](https://github.com/pocketbase/js-sdk) (internally it uses [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)).

Make sure that your request satisfies the `users` collection "Update API" rule (by default only admins and the authenticated user owner can edit the record).

To authenticate you need to send the `Authorization: TOKEN` header.

Additionally please note that the email can't change directly by "regular" users. You can find more information for the available body parameters if you scroll down in the same preview panel section in the "Body parameters" table.

**Comment by aristotekean** [comment]
Hi @ganigeorgiev  thanks for your answer, the example code its the same in documentation

  async editUser(data: any) {
       const record = await pb.collection('users').update(this.userId, data);
       console.log(record);
  } 
  
![image](https://user-images.githubusercontent.com/12689339/214066465-30914fd6-8626-4ec8-ac97-b09c50af49bc.png)

![image](https://user-images.githubusercontent.com/12689339/214066323-529ebcf2-e2b4-4818-a5a5-b910d4d1bec2.png)

![image](https://user-images.githubusercontent.com/12689339/214066719-63461715-a93e-42b0-a8e9-33036865cdd5.png)

 Only if I delete the following rule works
![image](https://user-images.githubusercontent.com/12689339/214066125-94c8ea05-4ca8-4651-81e3-cff22fdb96b0.png)

I try with angular http client and postman but the result its the same
<img width="725" alt="image" src="https://user-images.githubusercontent.com/12689339/214069294-66e0047f-c421-4d09-aad3-2c45047b0c8e.png">

**Comment by ganigeorgiev** [comment, marked as answer]
@aristotekean Are you sure that you are authorized? Just for the test you can try leaving the "Update API" rule empty.

Also note that in the Postman screenshot the `Authorization` header has unnecessary `TOKEN` prefix. Remove it and try again (aka. the value must be just the token).

**Reply by aristotekean** [reply]
Thanks, all errors was produced by the 'TOKEN' prefix I delete it from the request and works fine
